.github/workflows/coverity-scan.yaml

```diff
@@ -0,0 +1,45 @@
+# Tests in this workflow are run periodically.
+# They are generally very expensive and not on the critical path
+# for development
+# scheduled workflows only run in the default branch
+
+name: Coverity Scan
+
+on:
+  schedule:
+    - cron: "42 3 * * 0" # once a week
+  # allows to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  coverity-scan:
+    name: Scan with Coverity
+    # only run the workflow on Squid's main repository
+    if: github.repository == 'squid-cache/squid'
+
+    runs-on: ubuntu-22.04
+    # this job only runs if the github repository contains username
+    # and password to access the Coverity Scan service
+    env:
+      coverity_user: ${{ secrets.COVERITY_USER }}
+      coverity_token: ${{ secrets.COVERITY_TOKEN }}
+
+    container:
+      image: squidcache/buildfarm-coverity:stable
```

Comment:
If we really have to run cov-build in a container, please update PR description to explain why. If we can install the necessary packages/settings inside this workflow (without heroic efforts), like we do for functionality tests (among others), then let's do that instead of using a container.

Reply:
we would have to use heroic efforts: Coverity is not distributing their packages in a convenient format

Reply:
> we would have to use heroic efforts: Coverity is not distributing their packages in a convenient format

Coverity [documentation](https://community.blackduck.com/s/article/Coverity-Integrations-GitHub-with-GitHub-Hosted-Runners) for GitHub Actions integration says that it takes two curl and one cov-configure commands to install their scanner for Java code:

```yaml
    - name: Coverity Download
      run: |
        curl ... $COV_URL/downloadFile.htm?fn=$CSA.tar.gz | tar -C /tmp -xzf -
        curl ... -o /tmp/$CSA/bin/license.dat $COV_URL/downloadFile.htm?fn=license.dat
        /tmp/$CSA/bin/cov-configure --java
```

Do we really need a lot more than that for installing a C++ scanner?

Reply:
> Yes. That instruction is for a different product. Coverity also offers a GitHub action, but as I learnt from their support, it’s for their enterprise product and won’t work for the free version we use

Understood. I have addressed this change request by updating PR description accordingly. Please review and adjust further as needed.